test/core_tests.js

```diff
@@ -1,138 +1,156 @@
-    assert.ok( !step2.classList.contains("past"),
-               event.target.id + " unset past css class." );
-    done();
+(function() {
+  var registerEventListener = function( target, eventType, callback ) {
+    target.addEventListener( eventType, callback );
+    return function removeRegisteredEventListener() {
+      target.removeEventListener( eventType, callback );
+    };
   };
-  
-  root.addEventListener( "impress:stepenter",  function(event) { 
-                                                   setTimeout( function() { assertStepEnter( event ) }, wait );
-                                               } 
-  );
-
-  // Do no-op tests first, then trigger a transition. //
-
-  assert.strictEqual( impress().goto(document.querySelector( "div#impress" )),
-                    false,
-                    "goto() to a non-step element fails, as it should." );
-  assert.strictEqual( impress().goto(),
-                    false,
-                    "goto(<nothing>) fails, as it should." );
-
-  assert.ok( impress().next(),
-             "next() called and returns ok (1->2)" );
-  assert.ok( step2.classList.contains("active"),
-             step2.id + " set active css class." );
-  assert.ok( !step1.classList.contains("active"),
-             step1.id + " unset active css class." );      
-});
-
-
-// Cleanup
-QUnit.done(function( details ) {
-  // Impress.js will set the hash part of the url, we want to unset it when finished
-  // Otherwise a refresh of browser page would not start tests from step 1
-  window.location.hash = "";    
-  // Add back vertical scrollbar so we can read results if there were failures. 
-  document.body.style.overflow = 'auto';
-});
 
+  QUnit.begin(function() {
+    impress().init();
+  });
+
+  QUnit.module( "Initialization" );
+
+    QUnit.test( "Global Scope", function( assert ) {
+      assert.expect( 1 );
+      assert.ok( impress, "impress declared in global scope" );
+    });
+
+    QUnit.test( "Multiple API instantiation", function( assert ) {
+      assert.expect( 0 );
+      impress().init();
+    });
+
+    QUnit.test( "Support Markup", function( assert ) {
+      assert.expect( 4 );
+
+      var impressNotSupported = document.body.classList.contains( "impress-not-supported" );
+      var impressSupported = document.body.classList.contains( "impress-supported" );
+      assert.ok( impressSupported, "Have class .impress-supported" );
+      assert.notOk( impressNotSupported, "Don't have class .impress-not-supported" );
+
+      var impressDisabled = document.body.classList.contains( "impress-disabled" );
+      var impressEnabled = document.body.classList.contains( "impress-enabled" );
+      assert.ok( impressEnabled, "Have class .impress-enabled" );
+      assert.notOk( impressDisabled, "Don't have class .impress-disabled" );
+    });
+
+    QUnit.test( "Attributes", function( assert ) {
+      assert.expect( 9 );
+
+      var actual, expected;
```

Comment:
The actual/expected variable declaration is necessary because QUnit’s API by default does not name its argument. This problems falls in the same category as the [Boolean Trap](http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html). See [here](https://github.com/impress/impress.js/pull/510/files#r55665654).

Also, we declare variables on top to make it clear that it is being reused in this whole function scope.